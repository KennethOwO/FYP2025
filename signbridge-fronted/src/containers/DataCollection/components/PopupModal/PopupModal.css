.popup-modal {
    width: 40%;
    background: #fff;
    border-radius: 10px;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    text-align: center;
    padding: 0 30px 30px;
    color: #333;
    box-shadow: 2px 5px 10px rgba(0, 0, 0, 0.5);
    display: none;
    z-index: 1001;
}

.popup-modal.open {
    display: block;
}


.popup-tick {
    width: 20%;
    height: auto;
    margin-top: -50px;
    border-radius: 50%;
    box-shadow: 0 2px 5px rgba(0, 0, 0, 0.2);
}

.popup-modal h2 {
    font-size: 38px;
    font-weight: 500;
    margin: 30px 0 10px;
}

.popup-modal p {
    font-size: 22px;
    margin-top: 20px;
}

.popup-modal button {
    width: 100%;
    margin-top: 50px;
    padding: 10px 0;
    background-color: #6fd649;
    color: #fff;
    border: 0;
    outline: none;
    font-size: 25px;
    border-radius: 4px;
    cursor: pointer;
    box-shadow: 0 2px 5px rgba(0, 0, 0, 0.2);
}

@media (min-width: 641px) and (max-width: 768px) {
    .popup-tick {
        width: 40%;
        margin-top: -40px
    }
}

/* For devices with a max-width of 480px (small screens) */
@media (min-width: 320px) and (max-width: 480px) {
    .popup-modal {
        width: 90%;
        padding: 0 15px 20px;
    }

    .popup-tick {
        width: 30%;
    }

    .popup-modal h2 {
        font-size: 28px;
    }

    .popup-modal p {
        font-size: 16px;
    }

    .popup-modal button {
        font-size: 18px;
        padding: 8px 0;
    }
}

/* For devices with a width between 481px and 640px (mobile) */
@media (min-width: 481px) and (max-width: 640px) {
    .popup-modal {
        width: 80%;
        padding: 0 20px 25px;
    }

    .popup-tick {
        width: 25%;
    }

    .popup-modal h2 {
        font-size: 32px;
    }

    .popup-modal p {
        font-size: 18px;
    }

    .popup-modal button {
        font-size: 20px;
        padding: 9px 0;
    }
}

@media (min-width: 641px) and (max-width: 768px) {
    .popup-modal {
        width: 80%;
        padding: 0 20px 25px;
    }

    .popup-tick {
        width: 25%;
    }

    .popup-modal h2 {
        font-size: 32px;
    }

    .popup-modal p {
        font-size: 18px;
    }

    .popup-modal button {
        font-size: 20px;
        padding: 9px 0;
    }
}

/* For devices with a width between 769px and 1024px (tablets) */
@media (min-width: 769px) and (max-width: 1024px) {
    .popup-modal {
        width: 70%;
        padding: 0 25px 30px;
    }

    .popup-tick {
        width: 20%;
    }

    .popup-modal h2 {
        font-size: 34px;
    }

    .popup-modal p {
        font-size: 20px;
    }

    .popup-modal button {
        font-size: 22px;
        padding: 10px 0;
    }
}

/* For devices with a width between 1025px and 1280px (small desktops) */
@media (min-width: 1025px) and (max-width: 1280px) {
    .popup-modal {
        width: 50%;
    }

    .popup-tick {
        width: 18%;
    }

    .popup-modal h2 {
        font-size: 36px;
    }

    .popup-modal p {
        font-size: 20px;
    }

    .popup-modal button {
        font-size: 24px;
    }
}

/* For devices with a width between 1281px and 1440px (small desktops) */
@media (min-width: 1281px) and (max-width: 1440px) {
    .popup-modal {
        width: 40%;
    }

    .popup-tick {
        width: 15%;
    }

    .popup-modal h2 {
        font-size: 38px;
    }

    .popup-modal p {
        font-size: 22px;
    }

    .popup-modal button {
        font-size: 25px;
    }
}