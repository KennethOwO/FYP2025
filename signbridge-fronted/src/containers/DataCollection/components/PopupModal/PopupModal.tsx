import React from "react";
import "./PopupModal.css";
import image from "/images/tick2.png";
import { useTranslation } from "react-i18next";

interface PopupModalProps {
    isOpen: boolean;
    onClose: () => void;
}

const PopupModal: React.FC<PopupModalProps> = ({ isOpen, onClose }) => {
    const { t, i18n } = useTranslation();
    const handleConfirmClick = () => {
        onClose();
    };

    return (
        <div
            className={`popup-modal ${
                isOpen ? "open" : ""
            } modal-dialog-centered modal-lg`}
        >
            <img src={image} alt="" className="popup-tick img-fluid" />
            <div className="modal-content">
                <h2 className="modal-title">{t("insThankYou")}</h2>
                <p className="modal-body">{t("insProceedDetails")}</p>
                <button
                    className="btn btn-success"
                    onClick={handleConfirmClick}
                >
                    OK
                </button>
            </div>
        </div>
    );
};

export default PopupModal;
