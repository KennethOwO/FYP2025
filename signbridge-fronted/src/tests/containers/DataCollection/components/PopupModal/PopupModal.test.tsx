import React from "react";
import PopupModal from "../../../../../containers/DataCollection/components/PopupModal/PopupModal";
import { cleanup } from "@testing-library/react";
import { create } from "react-test-renderer";

jest.mock("react-i18next", () => ({
  useTranslation: () => ({
    t: (key: string) => key, // Mock the translation function to return the key
    i18n: {
      changeLanguage: jest.fn(), // Mock the changeLanguage function
    },
  }),
}));

describe("Test PopupModal", () => {
  afterEach(() => {
    // Clean up the DOM
    cleanup();
    // Reset mocks to their initial state
    jest.resetAllMocks();
  });
  const props = {
    isOpen: true,
    onClose: jest.fn(),
  };
  it("should render correctly", () => {
    const tree = create(<PopupModal {...props} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
