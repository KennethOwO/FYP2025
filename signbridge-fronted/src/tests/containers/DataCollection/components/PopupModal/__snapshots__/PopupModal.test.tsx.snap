// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PopupModal should render correctly 1`] = `
<div
  className="popup-modal open modal-dialog-centered modal-lg"
>
  <img
    alt=""
    className="popup-tick img-fluid"
    src="default"
  />
  <div
    className="modal-content"
  >
    <h2
      className="modal-title"
    >
      insThankYou
    </h2>
    <p
      className="modal-body"
    >
      insProceedDetails
    </p>
    <button
      className="btn btn-success"
      onClick={[Function]}
    >
      OK
    </button>
  </div>
</div>
`;
